sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1102,13 +1097,63 @@ def get_cmis_dp_init_duration_secs(self, api):
     def get_cmis_dp_deinit_duration_secs(self, api):
         return api.get_datapath_deinit_duration()/1000
 
-    def is_cmis_application_update_required(self, api, channel, speed):
+    def get_cmis_host_lanes(self, api, host_lane_count, channel, speed):
+        """
+        Retrieves bitmask of host lanes based on host lane count, channel and speed
+
+        Args:
+            api:
+                XcvrApi object
+            host_lane_count:
+                Integer, number of lanes on the host side
+            channel:
+                Integer, channel id of the group which the host lanes belong to (1 based)
+            speed:
+                Integer, the port speed of the host interface
+
+        Returns:
+            Integer, a bitmask of the lanes on the host side
+            e.g. 0x5 for lane 0 and lane 2.
+        """
+        host_lanes = 0
+
+        if channel > host_lane_count:
+            self.log_error("Failed to get host lanes as channel > host_lane_count!")
+            return host_lanes
+
+        if channel != 0:
+            appl = self.get_cmis_application_desired(api, host_lane_count, speed)
```

Comment:
I dont see we are iterating through all advertised 'appl' to find out if the host lane assignment satisfies the requested lane mapping and speed